lib/user.rb

```diff
@@ -0,0 +1,53 @@
+class User < Person
+  
+  attr_accessor :name, :gender, :age, :groceries
+
+  def initialize(args)
+    super
+  end
+
+  def address_person
+    if over_55?
+      grandparent = male? ? "grandfather" : "grandmother"
+      puts "Are you a #{grandparent}? [Y/N]"
+      spawn = gets.chomp.capitalize
+      puts "Aww, congratulations on your big genetic footprint!" if spawn == "Y"
+    elsif under_18?
+      gender_informal = male? ? "boy" : "girl"
+      puts "You're a young #{gender_informal}! Are you still in school? [Y/N]"
+      study = gets.chomp.capitalize
+      puts "I wish I could go to school like you." unless study == "N"
+    else
+      greeting = male? ? "man" : "woman"
+      pronoun = greeting == "man" ? "his" : "her"
```

Comment:
I would use `male?` here again instead of `greeting == "man"` -- increase flexibility.